include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -71,11 +76,10 @@ inline void InitApi() { Global<void>::api_ = OrtGetApiBase()->GetApi(ORT_API_VER
 const OrtApi* Global<T>::api_ = OrtGetApiBase()->GetApi(ORT_API_VERSION);
 #endif
 
-// This returns a reference to the OrtApi interface in use, in case someone wants to use the C API functions
+/// This returns a reference to the OrtApi interface in use
 inline const OrtApi& GetApi() { return *Global<void>::api_; }
 
-// This is a C++ wrapper for GetAvailableProviders() C API and returns
-// a vector of strings representing the available execution providers.
+/// This is a C++ wrapper for OrtApi::GetAvailableProviders() and returns a vector of strings representing the available execution providers.
 std::vector<std::string> GetAvailableProviders();
 
 // This is used internally by the C++ API. This macro is to make it easy to generate overloaded methods for all of the various OrtRelease* functions for every Ort* type
```

Comment:
>This is used internally by the C++ API [](http://example.com/codeflow?start=3&length=38)

if it's only used internally, can we #undef it in this header too?

Reply:
I'm not sure what you mean by #undef it. The header needs it to work. The users of the header can ignore it.

Reply:
I mean undefine it after it is used so the users don't need to ignore it

Reply:
Sure, we can since it's scoped to a small section. We don't do it in general though. I don't think users have to do any active work to ignore it...?